python/whylogs/api/logger/experimental/logger/actor/process_actor.py

```diff
@@ -45,6 +57,13 @@ def __init__(
         else:
             raise ValueError(f"Unknown queue type: {queue_type}")
 
+        self._sync_enabled = sync_enabled
+
+        # TODO is Any the right type here? It's the type of the PipeSignaler's message
+        # but BaseProcessRollingLogger appears to need some different things to work
+        # properly
+        self._pipe_signaler: Optional[PipeSignaler[Any]] = PipeSignaler() if self._sync_enabled is True else None
+
```

Comment:
The problem with the pipe signaler in the context of the base logger is that we use the same signaler to signal multiple types of things IIRC. One for making log synchronous, and another for the status check. The most correct type to use here is probably a union of w/e it is we signal for those two methods.

Reply:
I've tried with an union of both StatusType + MessageType, but it leads to errors on BaseLogger since it also has a case for a Future[None] type (L#433 on base_process_rolling_logger.py). then maybe `Optional[Union[ProcessStatus, ProcessMessageType]]` could work? idk, I saw you were using `[Any]` so I've decided to maintain

Reply:
I've tested and even with the Optional type, pyright complains about the Future[None] on BaseProcessingLogger. thoughts?

Reply:
Updated this branch with the fix